include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -424,7 +424,11 @@ struct __sycl_usm_alloc
     operator()(::std::size_t __elements) const
     {
         const auto& __queue = __exec.queue();
-        return (_T*)sycl::malloc(sizeof(_T) * __elements, __queue.get_device(), __queue.get_context(), __alloc_t);
+        auto __buf = static_cast<_T*>(
+            sycl::malloc(sizeof(_T) * __elements, __queue.get_device(), __queue.get_context(), __alloc_t));
+        if (!__buf)
+            throw std::bad_alloc();
```

Comment:
Should we include `<new>` header file ?

Reply:
> Should we include `<new>` header file ?

This should be included by `utils.h`.

> And the second question: if we are unable to allocate memory, should we use `sycl::buffer` instead?

I think we should instead check if USM is supported before using this routine. I think the only place where we currently use it is for reduce with heterogeneous backends and there we check it. This patch is mostly to handle out of resource errors.

Reply:
> I think we should instead check if USM is supported before using this routine. I think the only place where we currently use it is for reduce with heterogeneous backends and there we check it. This patch is mostly to handle out of resource errors.

OK.